File: src/main/java/com/nurkiewicz/tsclass/parser/visitors/MemberVariableDeclarationVisitor.java
package com.nurkiewicz.tsclass.parser.visitors;

import com.nurkiewicz.tsclass.antlr.parser.TypeScriptBaseVisitor;
import com.nurkiewicz.tsclass.antlr.parser.TypeScriptParser;
import com.nurkiewicz.tsclass.parser.ast.Field;

class MemberVariableDeclarationVisitor extends TypeScriptBaseVisitor<Field> {

    @Override
    public Field visitMemberVariableDeclaration(TypeScriptParser.MemberVariableDeclarationContext ctx) {
        final String name = ctx.variableDeclaration().IDENT().getSymbol().getText();
        final String type = ctx.variableDeclaration().typeAnnotation().type().getText();
        return new Field(name, type);
    }

}

File: src/main/kotlin/com/nurkiewicz/tsclass/parser/visitors/MemberVariableDeclarationVisitor.kt
package com.nurkiewicz.tsclass.parser.visitors

import com.nurkiewicz.tsclass.antlr.parser.TypeScriptBaseVisitor
import com.nurkiewicz.tsclass.antlr.parser.TypeScriptParser
import com.nurkiewicz.tsclass.parser.ast.Field
import com.nurkiewicz.tsclass.parser.ast.Type

internal class MemberVariableDeclarationVisitor : TypeScriptBaseVisitor<Field>() {

    override fun visitMemberVariableDeclaration(ctx: TypeScriptParser.MemberVariableDeclarationContext): Field {
        val name = ctx.variableDeclaration().IDENT().symbol.text
        val type = Type(ctx.variableDeclaration().typeAnnotation().type().text)
        return Field(name, type)
    }

}
